lib/graphlient/adapters/http/faraday_multipart_adapter/format_multipart_variables.rb

```diff
@@ -0,0 +1,60 @@
+require 'faraday'
+require 'mime/types'
+require 'graphlient/errors'
+
+module Graphlient
+  module Adapters
+    module HTTP
+      class FaradayMultipartAdapter
+        class NoMimeTypeException < Graphlient::Errors::Error; end
+        # Converts deeply nested File instances to Faraday::UploadIO
+        class FormatMultipartVariables
+          def initialize(variables)
+            @variables = variables
+          end
+
+          def call
+            deep_transform_values(variables) do |variable|
+              variable_value(variable)
+            end
+          end
+
+          private
+
+          attr_reader :variables
+
+          def deep_transform_values(hash, &block)
+            return hash unless hash.is_a?(Hash)
+
+            hash.transform_values do |val|
+              if val.is_a?(Hash)
+                deep_transform_values(val, &block)
+              else
+                yield(val)
+              end
+            end
+          end
+
+          def variable_value(variable)
+            if variable.is_a?(Array)
+              variable.map { |it| variable_value(it) }
+            elsif variable.is_a?(Hash)
+              variable.transform_values { |it| variable_value(it) }
+            elsif variable.is_a?(File)
+              file_variable_value(variable)
+            else
+              variable
+            end
+          end
+
+          def file_variable_value(file)
+            content_type = MIME::Types.type_for(file.path).first
+            return Faraday::UploadIO.new(file.path, content_type) if content_type
```

Comment:
should we use `Faraday::FilePart` or `Faraday::ParamPart` based on [this doc](https://github.com/lostisland/faraday/blob/80ba52c7217deee98e79dd023eb840bd2da30a82/docs/middleware/request/multipart.md)?

Reply:
Mhm. Good point. `Faraday::UploadIO` is deprecated. :disappointed: 
Thanks!